accumulator_updater/programs/accumulator_updater/src/state/accumulator_input.rs

```diff
@@ -14,26 +14,94 @@ use {
 /// the CPI calling program (e.g. Pyth Oracle)
 ///
 /// TODO: implement custom serialization & set alignment
-#[account]
+#[account(zero_copy)]
+#[derive(Debug, InitSpace)]
 pub struct AccumulatorInput {
     pub header: AccumulatorHeader,
-    pub data:   Vec<u8>,
+    // 10KB - 8 (discriminator) - 514 (header)
+    // TODO: do we want to initialize this to the max size?
+    //   - will lead to more data being passed around for validators
+    pub data:   [u8; 9_718],
 }
 
+//TODO:
+// - implement custom serialization & set alignment
+// - what other fields are needed?
+#[zero_copy]
+#[derive(InitSpace, Debug)]
+pub struct AccumulatorHeader {
+    pub bump:        u8, // 1
+    pub version:     u8, // 1
+    // byte offset of accounts where data starts
+    // e.g. account_info.data[offset + header_len]
+    pub header_len:  u16, // 2
+    /// endpoints of every message.
+    /// ex: [10, 14]
+    /// => msg1 = data[(header_len + 0)..(header_len + 10)]
+    /// => msg2 = data[(header_len + 10)..(header_len + 14)]
```

Comment:
👍 really like the examples here. they add a lot of clarity as to how this works

one note here is that you may want to rename "data" on AccumulatorInput because it can be confusing right now as to whether you're talking about the account_info.data or the accumulator_input.data

Reply:
renamed to messages